### PR TITLE
Refactor loki query and syntax hooks and remove duplicate hook call in editors

### LINKED ISSUE #21669: Loki: Simplify props on LokiQueryField and remove duplicate useLokiSyntax call
We use `LokiQueryField` inside `LokiQueryEditor` but we still call `useLokiSyntax` in both of them. This is because the props of `LokiQueryField` seem to require those syntax props even though it is creating them itself.

### PR DESCRIPTION
**Pull request by aocenas**
Closes: https://github.com/grafana/grafana/issues/21669

- Fixes prop types for some Loki editors so they do not need to duplicate calls to Loki syntax hook.
- Refactors the syntax hook a bit to separate syntax, labels and languageProvider initialisation.

### CODE REVIEW

File: public/app/plugins/datasource/loki/components/LokiQueryFieldForm.tsx [hunk 1]
```diff
@@ -73,8 +73,6 @@ export interface LokiQueryFieldFormProps extends ExploreQueryFieldProps<LokiData
 
 export class LokiQueryFieldForm extends React.PureComponent<LokiQueryFieldFormProps> {
   plugins: Plugin[];
-  modifiedSearch: string;
```
Comment (aocenas) [inline public/app/plugins/datasource/loki/components/LokiQueryFieldForm.tsx]: Wasn't used anywhere.

File: public/app/plugins/datasource/loki/components/useLokiLabels.ts [hunk 2]
```diff
@@ -9,22 +9,26 @@ import { useRefMounted } from 'app/core/hooks/useRefMounted';
...
   languageProviderInitialised: boolean,
-  activeOption: CascaderOption[],
   absoluteRange: AbsoluteTimeRange
 ) => {
   const mounted = useRefMounted();
 
   // State
   const [logLabelOptions, setLogLabelOptions] = useState([]);
   const [shouldTryRefreshLabels, setRefreshLabels] = useState(false);
-  const [shouldForceRefreshLabels, setForceRefreshLabels] = useState(false);
```
Comment (aocenas) [inline public/app/plugins/datasource/loki/components/useLokiLabels.ts]: Was only set to `false` never to `true`.

File: public/app/plugins/datasource/loki/components/useLokiSyntax.ts [hunk 3]
```diff
@@ -1,60 +0,0 @@
-import { useState, useEffect } from 'react';
```
Comment (aocenas) [inline public/app/plugins/datasource/loki/components/useLokiSyntax.ts]: Was renamed.

File: public/app/plugins/datasource/loki/components/useLokiSyntaxAndLabels.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
+import { useState, useEffect } from 'react';
```
Comment (aocenas) [inline public/app/plugins/datasource/loki/components/useLokiSyntaxAndLabels.ts]: Was renamed from useLokiSyntax. Not sure why that breaks github diff like this.

**Review by kaydelaney — COMMENTED** [review]
Haven't tested this out yet, but changes look good!

**Review by kaydelaney — APPROVED** [review]
LGTM

**Review by ivanahuckova — APPROVED** [review]
I really like the changes. Great refactor. LGTM!

**Pull request merged**